generator/src/lib.rs

```diff
@@ -1695,81 +1758,78 @@ pub fn derive_setters(
+                        let set_size_stmt = if array_size.contains("ename:") || array_size.contains('*') {
+                            // c_size should contain the same minus `ename:`-prefixed identifiers
+                            let array_size = field.c_size.as_ref().unwrap_or(array_size);
+                            let c_size = convert_c_expression(array_size);
+                            let inner_type = field.inner_type_tokens();
+
+                            slice_param_ty_tokens = quote!([#inner_type; #c_size]);
+
+                            quote!()
+                        } else {
+                            let array_size_ident = Ident::from(array_size.to_snake_case().as_str());
+                            quote!(self.inner.#array_size_ident = #param_ident_short.len() as _;)
+                        };
+
+                        return Some(quote! {
+                            pub fn #param_ident_short(mut self, #param_ident_short: &'a #mutable #slice_param_ty_tokens) -> #name_builder<'a> {
+                                #set_size_stmt
+                                self.inner.#param_ident = #param_ident_short#ptr;
+                                self
                             }
-                        }
-                        let slice_param_ty_tokens = Term::intern(&slice_param_ty_tokens);
-                        let ptr = Term::intern(ptr);
-
-                        if let vkxml::ArrayType::Dynamic = array_type {
-                                return Some(quote!{
-                                    pub fn #param_ident_short(mut self, #param_ident_short: #slice_param_ty_tokens) -> #name_builder<'a> {
-                                        self.inner.#array_size_ident = #param_ident_short.len() as _;
-                                        self.inner.#param_ident = #param_ident_short#ptr;
-                                        self
-                                    }
-                                });
-                        }
+                        });
                     }
                 }
             }
 
-            if param_ty_string.starts_with("*const ") {
-                let slice_param_ty_tokens = "&'a ".to_string() + &param_ty_string[7..];
-                let slice_param_ty_tokens = Term::intern(&slice_param_ty_tokens);
+            if field.is_const {
+                let slice_param_ty_tokens = field.safe_type_tokens(quote!('a));
                 return Some(quote!{
-                        pub fn #param_ident_short(mut self, #param_ident_short: #slice_param_ty_tokens) -> #name_builder<'a> {
-                            self.inner.#param_ident = #param_ident_short;
-                            self
-                        }
+                    pub fn #param_ident_short(mut self, #param_ident_short: #slice_param_ty_tokens) -> #name_builder<'a> {
+                        self.inner.#param_ident = #param_ident_short;
+                        self
+                    }
```

Comment:
While refactoring I came across this case and wondered why it is not dealing with the mutable pointer case. This PR introduces `to_safe_tokens` that already "substitutes" `*` with `&'a` and the appropriate mutability keyword. It could be used but breaks some `*mut c_void` setters. Addressing that and it turns up there are quite a few `&'a c_void` in the codebase as well. What do we want to do here? See the change this results in here: https://github.com/MaikKlein/ash/commit/ffa0a76ef28c553946136dcadc60a5dbf4689028

Reply:
Hm good question, I am not sure.. I think we want to use pointers for any opaque types. IIRC `ANativeWindow` is one as well.

Reply:
I've since added `ANativeWindow` and `CAMetalLayer`, see the new commit here: 4d9647f0f8617393f2e26dbba0d94d91e309b50e

Now the only relevant mutable reference change is:
```diff
@@ -37491,7 +37491,7 @@ impl<'a> ::std::ops::DerefMut for PipelineCreationFeedbackCreateInfoEXTBuilder<'
 impl<'a> PipelineCreationFeedbackCreateInfoEXTBuilder<'a> {
     pub fn pipeline_creation_feedback(
         mut self,
-        pipeline_creation_feedback: *mut PipelineCreationFeedbackEXT,
+        pipeline_creation_feedback: &'a mut PipelineCreationFeedbackEXT,
     ) -> PipelineCreationFeedbackCreateInfoEXTBuilder<'a> {
         self.inner.p_pipeline_creation_feedback = pipeline_creation_feedback;
         self
```

And converting a bunch of `&'a <opaque/void type>` to `*const`.